### PR TITLE
Add missing fields in timer object.

### PR DESCRIPTION
**Pull request by pdthummar**
Added missing fields in timer object. 
Fixes: https://github.com/Azure/azure-functions-python-worker/issues/681
Fixes: https://github.com/Azure/azure-functions-python-worker/issues/858

Missing fields

### CODE REVIEW

File: azure/functions/timer.py [hunk 1]
```diff
@@ -1,23 +1,30 @@
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
```
Comment (gavin-aguiar) [inline azure/functions/timer.py:2]: Why is this removed?

File: azure/functions/timer.py [hunk 2]
```diff
@@ -32,5 +39,8 @@ def decode(cls, data: meta.Datum, *, trigger_metadata) -> typing.Any:
             raise NotImplementedError
 
         info = json.loads(data.value)
+
         return TimerRequest(
-            past_due=info.get('IsPastDue', False))
+            past_due=info.get('IsPastDue', False),
+            schedulestatus=info["ScheduleStatus"],
```
Comment (gavin-aguiar) [inline azure/functions/timer.py:45]: We should probably keep this the same as how we retrieve pastdue with a default value.  e.g info.get("ScheduleStatus" , {})

File: azure/functions/timer.py [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
         self.__past_due = past_due
+        self.__schedulestatus = schedulestatus
+        self.__schedule = schedule
 
     @property
     def past_due(self):
         return self.__past_due
 
+    @property
+    def schedulestatus(self):
```
Comment (gavin-aguiar) [inline azure/functions/timer.py:21]: snake case this, schedule_status

File: tests/test_timer.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import json
+import unittest
+
+import azure.functions.timer as timer
+from azure.functions.meta import Datum
+
+
+class TestTimer(unittest.TestCase):
+    def test_TimerRequestConverter_decode(self):
```
Comment (gavin-aguiar) [inline tests/test_timer.py:12]: We might need to add some more tests. Check the test cases in other triggers. 
e.g. test_timer_input_type
test_timer_output_type
test_timer_decode
test_timer_initialize_without_args
test_timer_initialize_all_args

**Review by gavin-aguiar — CHANGES_REQUESTED** [review]

File: azure/functions/timer.py [hunk 5]
```diff
@@ -1,23 +1,30 @@
...
+        return self.__schedulestatus
+
+    @property
+    def schedule(self):
```
Comment (vrdmr) [inline azure/functions/timer.py:25]: Can we also write the expected return types for all the properties?

File: azure/functions/timer.py (hunk 2 above)
Comment (vrdmr) [reply azure/functions/timer.py:45]: Yes - what if info doesn't have ScheduleStatus - it would throw key error. - we should take care of this.

**Review by vrdmr — CHANGES_REQUESTED** [review]

File: azure/functions/timer.py (hunk 1 above)
Comment (pdthummar) [reply azure/functions/timer.py]: It was mistakenly removed. I have added in new commit.

File: azure/functions/timer.py (hunk 3 above)
Comment (pdthummar) [reply azure/functions/timer.py]: updated this as per recommendation.

File: azure/functions/timer.py (hunk 2 above)
Comment (pdthummar) [reply azure/functions/timer.py]: I have updated this change as per recommendation.

File: azure/functions/timer.py (hunk 5 above)
Comment (pdthummar) [reply azure/functions/timer.py]: I have included expected return types in updated file. Thanks.

File: tests/test_timer.py (hunk 4 above)
Comment (pdthummar) [reply tests/test_timer.py]: I have added few more tests. thanks

File: azure/functions/timer.py [hunk 6]
```diff
@@ -5,19 +5,29 @@
...
-    def __init__(self, *, past_due: bool) -> None:
+    def __init__(self, *, past_due: bool = False, schedule_status: dict = {},
+                 schedule: dict = {}) -> None:
         self.__past_due = past_due
+        self.__schedule_status = schedule_status
+        self.__schedule = schedule
 
     @property
-    def past_due(self):
+    def past_due(self) -> bool:
```
Comment (vrdmr) [inline azure/functions/timer.py:20]: 😍  Thanks a lot for this.

File: azure/functions/timer.py [hunk 7]
```diff
@@ -5,19 +5,29 @@
 import typing
 
 from azure.functions import _abc as azf_abc
-
 from . import meta
 
 
 class TimerRequest(azf_abc.TimerRequest):
```
Comment (vrdmr) [inline azure/functions/timer.py:11]: @YunchuWang - Getting your input here - If we plan to add PyDocs here - what could we add here?

@pdthummar No need to worry in this PR (mostly). But we should explore adding more PyDoc comments as they eventually show up in [azure.functions.timerrequest](https://docs.microsoft.com/en-us/python/api/azure-functions/azure.functions.timerrequest?view=azure-python)

File: tests/test_timer.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
```
Comment (vrdmr) [inline tests/test_timer.py:2]: Curious - weren't there any Timer tests before this? 😱

File: tests/test_timer.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+import json
+import unittest
+
+import azure.functions.timer as timer
+from azure.functions.meta import Datum
+
+
+class TestTimer(unittest.TestCase):
+    def test_timer_decode(self):
+        # given
```
Comment (vrdmr) [inline tests/test_timer.py:13]: I liked the given-when-then structure you used. 😃  👍🏼

**Review by vrdmr — APPROVED** [review]
🚢  
LGTM

File: tests/test_timer.py (hunk 8 above)
Comment (pdthummar) [reply tests/test_timer.py:2]: there were not any timer tests before.

**Review by gavin-aguiar — APPROVED** [review]
LGTM

File: azure/functions/timer.py (hunk 7 above)
Comment (YunchuWang) [reply azure/functions/timer.py:11]: @vrdmr  would suggest giving a overview what this class does, adding definition of each field in init, and link to https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=in-process&pivots=programming-language-java#attributes

**Comment by pdthummar** [comment]
@lak-amith531 If you are using old Coretools version, Can you please upgrade it to latest version and try it? It should work with latest version. 

If you are still facing issue after upgrading version, please create an issue with including replication steps. Thanks

**Comment by tim-sparq** [comment]
this doesn't work when schedule is more frequent than 1x p/min, even when useMonitor is set to true in function.json
